homeassistant/components/mqtt/image.py

```diff
@@ -0,0 +1,199 @@
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_IMAGE_ENCODING): "b64",
+        vol.Optional(CONF_CONTENT_TYPE, default=DEFAULT_CONTENT_TYPE): vol.All(
+            cv.string, validate_content_type
+        ),
+        vol.Optional(CONF_FROM_URL, default=False): cv.boolean,
+    }
+).extend(MQTT_ENTITY_COMMON_SCHEMA.schema)
+
+PLATFORM_SCHEMA_MODERN = vol.All(
+    PLATFORM_SCHEMA_BASE.schema,
+)
+
+DISCOVERY_SCHEMA = PLATFORM_SCHEMA_BASE.extend({}, extra=vol.REMOVE_EXTRA)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up MQTT image through YAML and through MQTT discovery."""
+    setup = functools.partial(
+        _async_setup_entity, hass, async_add_entities, config_entry=config_entry
+    )
+    await async_setup_entry_helper(hass, image.DOMAIN, setup, DISCOVERY_SCHEMA)
+
+
+async def _async_setup_entity(
+    hass: HomeAssistant,
+    async_add_entities: AddEntitiesCallback,
+    config: ConfigType,
+    config_entry: ConfigEntry,
+    discovery_data: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the MQTT Image."""
+    async_add_entities([MqttImage(hass, config, config_entry, discovery_data)])
+
+
+class MqttImage(MqttEntity, ImageEntity):
+    """representation of a MQTT image."""
+
+    _entity_id_format: str = image.ENTITY_ID_FORMAT
+    _attributes_extra_blocked: frozenset[str] = MQTT_IMAGE_ATTRIBUTES_BLOCKED
+    _last_image: bytes | None = None
+    _client: httpx.AsyncClient
+    _template: Callable[[ReceivePayloadType], ReceivePayloadType]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
+        """Initialize the MQTT Image."""
+        self._client = get_async_client(hass)
+        ImageEntity.__init__(self)
+        MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
+
+    @staticmethod
+    def config_schema() -> vol.Schema:
+        """Return the config schema."""
+        return DISCOVERY_SCHEMA
+
+    def _setup_from_config(self, config: ConfigType) -> None:
+        """(Re)Setup the entity."""
+        if not config[CONF_FROM_URL]:
+            self._attr_content_type = config[CONF_CONTENT_TYPE]
+        self._template = MqttValueTemplate(
+            config.get(CONF_VALUE_TEMPLATE), entity=self
+        ).async_render_with_possible_json_value
+
+    async def _async_load_image(self, url: str) -> None:
+        try:
+            response = await self._client.request("GET", url)
+        except (httpx.TimeoutException, httpx.RequestError, ssl.SSLError) as ex:
+            _LOGGER.warning("Connection failed to url %s files: %s", url, ex)
+            return
```

Comment:
Should we set the image to None if we fail?

Reply:
As far as I know we do not update the state in case of an error in with other MQTT platforms.

Reply:
Good point